File: single-language-analysis/dataset-files/python/test_smell_046.py
import pytest

class TestUhu:
    def test_smell_046(self):
        a = Uhu()
        food = ""
        food_list_correct = True
        bad = ""
        for i in range(len(DIET_ITEMS)):
            food = DIET_ITEMS[i]
            try:
                a.eat(food)
            except FishFoodException as fe:
                food_list_correct = False
                bad = food
            except FishSizeException as fs:
                pass
        
        assert food_list_correct, "Fish diet items are not correct. " + bad + " is not a diet item"

File: single-language-analysis/dataset-files/java/TestSmell046.java
// Original URL: https://github.com/ICSatKCC/a6-lawa-i-fishing-game-f23-g4_f23_a6/blob/71a87ba82a51202deb39bc625d80350912af1dbb/UhuTest.java#L117-L137

public class TestSmell046 {
   @Test public void eatingDietTest(){
      I_a a = new Uhu();
      String food = "";
      boolean foodListCorrect = true;
      String bad = "";
      for(int i = 0; i < DIET_ITEMS.length; i++){
         food = DIET_ITEMS[i];
         try{
            a.eat(food);
         } catch(FishFoodException fe) {
            foodListCorrect = false;
            bad = food;
         } catch(FishSizeException fs) {
            //do nothing food ok but fish got too big
         }
      
      }
      Assert.assertTrue("Fish diet items are not correct. " + bad 
         + " is not a diet item", foodListCorrect);

   } //eating test
}
